_posts/05-06-01-Internationalization-and-Localization.md

```diff
@@ -0,0 +1,348 @@
+
+Now that you understood the basis of how plural rules works - and if you didn't, please look at a deeper explanation
+on the [LingoHub tutorial](lingohub_plurals) -, you might want to copy the ones you need from a [list][plural] instead of
+writing them by hand.
+
+When calling out Gettext to do localization on sentences with counters, you'll have to give him the
+related number as well. Gettext will work out what rule should be in effect and use the correct localized version.
+You will need to include in the `.po` file a different sentence for each plural rule defined.
+
+### Sample implementation
+After all that theory, let's get a little practical. Here's an excerpt of a `.po` file - don't mind with its format,
+but instead the overall content, you'll learn how to edit it easily later:
+
+{% highlight po %}
+msgid ""
+msgstr ""
+"Language: pt_BR\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+
+msgid "We're now translating some strings"
+msgstr "Nós estamos traduzindo algumas strings agora"
+
+msgid "Hello %1$s! Your last visit was on %2$s"
+msgstr "Olá %1$s! Sua última visita foi em %2$s"
+
+msgid "Only one unread message"
+msgid_plural "%d unread messages"
+msgstr[0] "Só uma mensagem não lida"
+msgstr[1] "%d mensagens não lidas"
+{% endhighlight %}
+
+The first section works like a header, having the `msgid` and `msgstr` specially empty. It describes the file encoding,
+plural forms and other things that are less relevant.
+The second section translates a simple string from English to
+Brazilian Portuguese, and the third does the same, but leveraging string replacement from [`sprintf`](sprintf) so the
+translation may contain the user name and visit date.  
+The last section is a sample of pluralization forms, displaying
+the singular and plural version as `msgid` in English and their corresponding translations as `msgstr` 0 and 1
+(following the number given by the plural rule). There, string replacement is used as well so the number can be seen
+directly in the sentence, by using `%d`. The plural forms always have two `msgid` (singular and plural), so it's
+advised to not use a complex language as source of translation.
+
+### Discussion on l10n keys
+As you might have noticed, we're using as source ID the actual sentence in English. That `msgid` is the same used
+throughout all your `.po` files, meaning other languages will have the same format and the same `msgid` fields, but
+translated `msgstr` lines.
+
+Talking about translation keys, there are two main "schools" here:
+
+1. _`msgid` as a real sentence_.  
+    The main advantage are: 
+    - if there's pieces of the software untranslated in any given language, the key displayed will still maintain some
+    meaning. Example: if you happen to translate by heart from English to Spanish but needs help to translate to French,
```

Comment:
3 suggestions:

The main ~~advantage~~ `advantages` are: 
    - if ~~there's~~ `there are` pieces of the software untranslated in any given language, the key displayed will still maintain some
    meaning. Example: if you happen to translate by heart from English to Spanish but ~~needs~~ `need` help to translate to French,